### PR TITLE
HWKAPM-787 Fix issue where internal fragments incorrectly classified …

### PR DESCRIPTION
**Pull request by objectiser**
…as client invocations

### CODE REVIEW

**Comment by objectiser** [comment]
@pavolloffay Could you review please?

File: server/processors/src/test/java/org/hawkular/apm/server/processor/nodedetails/NodeDetailsDeriverTest.java [hunk 1]
```diff
@@ -137,4 +141,31 @@ public void testProcessCommonProperties() throws RetryAttemptException {
...
+
+        Producer producer = new Producer();
+        producer.setEndpointType("HTTP");
+        consumer.getNodes().add(producer);
+
+        List<NodeDetails> details = null;
+
+        try {
+            details = deriver.processOneToMany(null, trace);
+        } catch (Exception e) {
```
Comment (pavolloffay) [inline server/processors/src/test/java/org/hawkular/apm/server/processor/nodedetails/NodeDetailsDeriverTest.java]: This should be test error

File: server/processors/src/test/java/org/hawkular/apm/server/processor/nodedetails/NodeDetailsDeriverTest.java [hunk 2]
```diff
@@ -137,4 +141,31 @@ public void testProcessCommonProperties() throws RetryAttemptException {
         assertEquals(new HashSet<>(Arrays.asList(p1,p2,p3,p4)), details.get(2).getProperties());
     }
 
+    @Test
+    public void testProcessInternalFragmentWithProducer() {
```
Comment (pavolloffay) [inline server/processors/src/test/java/org/hawkular/apm/server/processor/nodedetails/NodeDetailsDeriverTest.java]: Maybe rename it to `testInitial` or something like that? As it is testing only that.

**Comment by objectiser** [comment]
@pavolloffay Thanks - PR updated.

**Comment by objectiser** [comment]
@pavolloffay Thanks!

**Pull request merged**